benchmarks/bench_plot_randomized_svd.py

```diff
@@ -0,0 +1,225 @@
+    plt.legend(loc="upper left")
+    plt.suptitle("Frobenius distance vs. noise\n\
+                 matrix shape = %i x %i, rank = %i, n comps in rand_SVD = %i" %
+                 (data.shape[0], data.shape[1], rank, n_comps))
+    plt.ylabel("Frobenius distance")
+    plt.xlabel("Noise component")
+
+
+def l2_singular_values_distance(n_power_iter, n_comps, X, true_mu=None,
+                                distance=squared_norm):
+    gc.collect()
+    t0 = time()
+    _, mu, _ = randomized_svd(X, n_comps, n_iter=n_power_iter)
+    call_time = time() - t0
+
+    if mu is None:
+        _, true_mu, _ = sp.linalg.svd(X, full_matrices=False)
+    d = distance(true_mu[:n_comps] - mu)
+    return {'time': call_time, 'distance': d}
+
+
+def frobenius_reconstruction_distance(method, X, n_power_iter=0, n_comps=0,
+                                      distance=squared_norm):
+    if method == 'randomized_SVD':
+        gc.collect()
+        t0 = time()
+        U, mu, V = randomized_svd(X, n_comps, n_iter=n_power_iter)
+        call_time = time() - t0
+    elif method == 'SVD':
+        gc.collect()
+        t0 = time()
+        U, mu, V = sp.linalg.svd(X, full_matrices=False)
+        call_time = time() - t0
+
+    d = distance(U.dot(np.diag(mu)).dot(V) - X)
+    return {'time': call_time, 'distance': d}
+
+
+def noisy_matrices_comparison(data, noise_list, power_iter_list, rank,
+                              n_comps):
+    """
+    Rational for this test: show space of improvements in terms of
+    approximation of singular values or reconstruction, if we play with
+    the number of power iterations.
+    """
+
+    all_l2 = defaultdict(list)
+    all_frobenius = defaultdict(list)
+    for i, n in enumerate(noise_list):
+        # Compute the exact singular values once per dataset
+        _, mu, _ = sp.linalg.svd(data[i], full_matrices=False)
+        for p in power_iter_list:
+            d_l2 = \
+                l2_singular_values_distance(p, n_comps, data[i],
+                                            true_mu=mu)['distance']
+            params = ['randomized_SVD', data[i], p, n_comps]
+            d_frobenius = \
+                frobenius_reconstruction_distance(*params)['distance']
+
+            all_l2[p].append(d_l2)
+            all_frobenius[p].append(d_frobenius)
+
+    plot_noise_vs_l2_distance(noise_list, all_l2, data[0], rank, n_comps)
+    plot_noise_vs_frobenius_distance(noise_list, all_frobenius,
+                                     data[0], rank, n_comps)
+
+
+def plot_time_vs_distance(time, l2, frobenius, p, p_comps, dataset_name):
+    plt.figure()
+    sizes = (3 + p) * 15
+    plt.scatter(time, l2, label='l2', marker='o', s=sizes, c='b')
+    plt.scatter(time, frobenius, label='Frobenius', marker='o', s=sizes, c='r')
+    plt.legend(loc="lower right")
+    plt.axhline(y=0, ls=':', c='black')
+    plt.suptitle("%s: distances from SVD vs. running time\n \
+                  fraction of compontens used in rand_SVD = %i" %
```

Comment:
Typo: compon**ent**s.

Also from your plots, it looks like this is the number of components and not the fraction right?